esp32_s3_lcd_ev_board/src/bsp_lvgl_port.c

```diff
@@ -0,0 +1,626 @@
+    disp_drv.direct_mode = 1;
+#endif
+    return lv_disp_drv_register(&disp_drv);
+}
+
+static void touchpad_read(lv_indev_drv_t *indev_drv, lv_indev_data_t *data)
+{
+    esp_lcd_touch_handle_t tp = (esp_lcd_touch_handle_t)indev_drv->user_data;
+    assert(tp);
+
+    uint16_t touchpad_x;
+    uint16_t touchpad_y;
+    uint8_t touchpad_cnt = 0;
+    /* Read data from touch controller into memory */
+    esp_lcd_touch_read_data(tp);
+
+    /* Read data from touch controller */
+    bool touchpad_pressed = esp_lcd_touch_get_coordinates(tp, &touchpad_x, &touchpad_y, NULL, &touchpad_cnt, 1);
+    if (touchpad_pressed && touchpad_cnt > 0) {
+        data->point.x = touchpad_x;
+        data->point.y = touchpad_y;
+        data->state = LV_INDEV_STATE_PRESSED;
+        ESP_LOGD(TAG, "Touch position: %d,%d", touchpad_x, touchpad_y);
+    } else {
+        data->state = LV_INDEV_STATE_RELEASED;
+    }
+}
+
+static lv_indev_t *indev_init(esp_lcd_touch_handle_t tp)
+{
+    BSP_NULL_CHECK(tp, NULL);
+
+    static lv_indev_drv_t indev_drv_tp;
+
+    /* Register a touchpad input device */
+    lv_indev_drv_init(&indev_drv_tp);
+    indev_drv_tp.type = LV_INDEV_TYPE_POINTER;
+    indev_drv_tp.read_cb = touchpad_read;
+    indev_drv_tp.user_data = tp;
+
+    return lv_indev_drv_register(&indev_drv_tp);
+}
+
+static void tick_increment(void *arg)
+{
+    /* Tell LVGL how many milliseconds have elapsed */
+    lv_tick_inc(LVGL_TICK_PERIOD_MS);
+}
+
+static esp_err_t tick_init(void)
+{
+    // Tick interface for LVGL (using esp_timer to generate 2ms periodic event)
+    const esp_timer_create_args_t lvgl_tick_timer_args = {
+        .callback = &tick_increment,
+        .name = "LVGL tick"
+    };
+    esp_timer_handle_t lvgl_tick_timer = NULL;
+    BSP_ERROR_CHECK_RETURN_ERR(esp_timer_create(&lvgl_tick_timer_args, &lvgl_tick_timer));
+    return esp_timer_start_periodic(lvgl_tick_timer, LVGL_TICK_PERIOD_MS * 1000);
+}
+
+static void lvgl_port_task(void *arg)
+{
+    ESP_LOGI(TAG, "Starting LVGL task");
+    while (1) {
+        bsp_display_lock(0);
+        uint32_t task_delay_ms = lv_timer_handler();
+        bsp_display_unlock();
+        if (task_delay_ms > 500) {
+            task_delay_ms = 500;
+        } else if (task_delay_ms < CONFIG_BSP_DISPLAY_LVGL_TASK_DELAY) {
+            task_delay_ms = CONFIG_BSP_DISPLAY_LVGL_TASK_DELAY;
+        }
+        vTaskDelay(pdMS_TO_TICKS(task_delay_ms));
+    }
+}
+
+esp_err_t bsp_lvgl_port_init(esp_lcd_panel_handle_t lcd, esp_lcd_touch_handle_t tp, lv_disp_t **disp, lv_indev_t **indev)
+{
+    BSP_NULL_CHECK(lcd, ESP_ERR_INVALID_ARG);
+    BSP_NULL_CHECK(tp, ESP_ERR_INVALID_ARG);
+    BSP_NULL_CHECK(disp, ESP_ERR_INVALID_ARG);
+    BSP_NULL_CHECK(indev, ESP_ERR_INVALID_ARG);
+
+    lv_init();
+    BSP_ERROR_CHECK_RETURN_ERR(tick_init());
+    BSP_NULL_CHECK(*disp = display_init(lcd), ESP_FAIL);
+    BSP_NULL_CHECK(*indev = indev_init(tp), ESP_FAIL);
+
+#if CONFIG_BSP_DISPLAY_LVGL_ROTATION_90
```

Comment:
Yeah, it's possible. But this rotation feature relies on the Anti-tearing function, which is currently controlled by macros. It might be better to implement both of them as dynamic configurations in `esp_lvgl_port` in the next version.